test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs

```diff
@@ -37,22 +37,22 @@ void PredictAndMetadata()
 
             var testLoader = ml.Data.LoadFromTextFile(dataPath, TestDatasets.irisData.GetLoaderColumns(), separatorChar: ',', hasHeader: true);
             var testData = ml.Data.CreateEnumerable<IrisData>(testLoader, false);
-            
+
             // During prediction we will get Score column with 3 float values.
             // We need to find way to map each score to original label.
-            // In order to do what we need to get SlotNames from Score column.
-            // Slot names on top of Score column represent original labels for i-th value in Score array.
-            VBuffer<ReadOnlyMemory<char>> slotNames = default;
-            engine.OutputSchema[nameof(IrisPrediction.Score)].GetSlotNames(ref slotNames);
+            // In order to do what we need to get TrainingLabelValues from Score column.
+            // TrainingLabelValues on top of Score column represent original labels for i-th value in Score array.
+            VBuffer<ReadOnlyMemory<char>> originalLabels = default;
```

Comment:
>ReadOnlyMemory [](start = 20, length = 14)

In this particular case we should be propagating *both* slot names and label names, right? Since they're string in both cases? While I see the point in augmenting the test to cover this new metadata type, is there any particular reason to remove the test that the vector has teh appropriate slot names?

Reply:
This is old scenario test. Purpose of it to show user how to do work with metadata.
In this particular test label is string, and it has slotnames, but it wont in case of non string label.
If I add slotnames here it would be confusing? What for do I get slotnames?

All tests with TestEstimatorCore routing would test on presence of slotnames and TrainingLabelValues. And we have plenty of them,. why should we do anything here with slotname?

---
In reply to: [271046772](https://github.com/dotnet/machinelearning/pull/3101#discussion_r271046772) [](ancestors = 271046772)

Reply:
OK, that's fine. I had the idea that these "showing the user" things were more the sake of functional tests, but as you like.

---
In reply to: [271050961](https://github.com/dotnet/machinelearning/pull/3101#discussion_r271050961) [](ancestors = 271050961,271046772)